package/MDAnalysis/core/groups.py

```diff
@@ -1778,13 +1820,35 @@ def n_segments(self):
         """Number of unique segments represented in the AtomGroup.
 
         Equivalent to ``len(self.segments)``.
-
         """
         return len(self.segments)
 
     @property
+    @cached('unique_restore_mask')
+    def _unique_restore_mask(self):
+        # The _unique_restore_mask property's cache is populated whenever the
+        # AtomGroup.unique property of a *non-unique* AtomGroup is accessed.
+        # If _unique_restore_mask is not cached, it is *definitely* used in the
+        # wrong place, so we raise an exception here. In principle, the
+        # exception should be an AttributeError, but the error message would
+        # then be replaced by the __getattr__() error message. To prevent the
+        # message from being overridden, we raise a RuntimeError instead.
+        if self.isunique:
```

Comment:
since this attribute is behind an underscore, is this check required?

Reply:
The body of this getter function should actually never be executed, since all methods requiring the attribute must have accessed `AtomGroup.unique` before, and thus, the data is read from cache. If the code works correctly, a user will *never* see anything of this. The different error messages are intended for developers only, and differentiate between the case where the attribute has been accessed from a unique or non-unique `AtomGroup` to help debugging the issue. Maybe the error messages should be more verbose in that respect.

Reply:
I'm not sure we really need error messages and extra code for attributes which aren't accessible, but I guess it doesn't hurt either